compiler/src/yul/mappers/expressions.rs

```diff
@@ -92,17 +92,21 @@ pub fn expr_call(
     exp: &Spanned<fe::Expr>,
 ) -> Result<yul::Expression, CompileError> {
     if let fe::Expr::Call { args, func } = &exp.node {
-        if let fe::Expr::Attribute { value: _, attr } = &func.node {
-            let arguments = &args.node;
-            let yul_args: Vec<yul::Expression> = arguments
-                .iter()
-                .map(|val| call_arg(context, val))
-                .collect::<Result<_, _>>()?;
-
-            let func_name = identifier! { (attr.node) };
-
-            return Ok(expression! { [func_name]([yul_args...]) });
-        }
+        return match &func.node {
```

Comment:
I dislike matching the actual AST node here, we should use a different data structure provided by semantic analysis...

Reply:
Yeah, I think the compiler books also say that the semantic pass would provide a new abstraction to the mapper pass. So far, I haven't seen how that would diverge in a meaningful way but I'm open to the idea :)